lib/pure/osproc.nim

```diff
@@ -759,6 +811,21 @@ elif not defined(useNimRtl):
          pipe(pStderr) != 0'i32:
         raiseOSError(osLastError())
 
+      if hIn != INVALID_HANDLE_VALUE:
+        discard close(pStdin[readIdx])
+        pStdin[readIdx] = hIn
```

Comment:
pStdin[readIdx], pStdout[writeIdx] and pStderr[writeIdx] are closed by the original code at the end of the proc so those shouldn't leak.